tests/system/src/test/java/test/util/ScreenCaptureTestWatcher.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2024, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.util;
+
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.imageio.ImageIO;
+import javafx.embed.swing.SwingFXUtils;
+import javafx.geometry.Rectangle2D;
+import javafx.scene.image.WritableImage;
+import javafx.scene.robot.Robot;
+import javafx.stage.Screen;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.TestWatcher;
+
+/**
+ * Standard Test Watcher for Headful Tests (JUnit 5 Only).
+ * <p>
+ * This facility takes a screenshot of any failed test, then logs the base64-encoded screenshot
+ * to {@code stderr}.
+ * <p>
+ * To use, simply add the following annotation to your class:
+ * <pre>{@code
+ * @ExtendWith(ScreenCaptureTestWatcher.class)
+ * }</pre>
+ * <p>
+ * For Eclipse users, override dependencies with the following entries:
+ * <pre>{@code
+ * --add-modules=javafx.base,javafx.graphics,javafx.controls,javafx.swing
+ * --add-opens javafx.controls/test.com.sun.javafx.scene.control.test=javafx.base
+ * --add-exports javafx.base/com.sun.javafx=ALL-UNNAMED
+ * -Djava.library.path="../../../../build/sdk/lib"
+ * -ea
+ * }</pre>
+ * <p>
+ * WARNING: using this utility may significantly increase the size of your logs!
+ * Make sure there is plenty of free disk space.
+ */
+// TODO investigate having a hard-coded or programmable via property limit on the number of
+// captured screenshots.
```

Comment:
I had the same thought. Given that each test runs in its own JVM, I can't think of a good way to do it other than recording information in a file in tests/system/build.

Reply:
Exactly.  I don't have a good way, and I would rather not write to a file.